### ISSUE TITLE
Right-click on track or button to add corresponding RESET animation

### ISSUE THREAD

**Issue by hsandt** [godotengine/godot-proposals#7548]
### Describe the project you are working on

A combat platformer

### Describe the problem or limitation you are having in your project

In the Animation panel, if I forgot to add a RESET track when I added an animation property (can happen when you use Add Track > select property instead of clicking on the key button near the property field, as first method won't prompt you to add RESET track), I want to add it later. But it's cumbersome to remember all the tracks to add and add them manually in the RESET animation.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Right-click on track, or click on some button nearby to add a corresponding entry in the RESET animation. We need some value for the RESET animation, I suggest to copy the current animation value on the track we clicked. If RESET animation already has this track, the command/button is greyed out.

Advanced: a single button to copy *all* tracks that are not registered for RESET yet to the RESET animation, picking their first key value or value at time 0 if any, else use default "zero" value.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

![Godot 4 - add track to RESET animation](https://github.com/godotengine/godot-proposals/assets/1067772/58265c15-6fd9-4d8e-a134-942ad4ddbfd6)

### If this enhancement will not be used often, can it be worked around with a few lines of script?

No, it's a core window

### Is there a reason why this should be core and not an add-on in the asset library?

Yes, it's a core window

**Comment by KoBeWi** [comment]
This exists, just right-click on the animation key.

**Comment by IceTyp** [comment]
It seems that this is only available when `AnimationTrackKeyEdit` is opened in the inspector. Otherwise, there is only `Insert Key`. (Same with `Duplicate Key(s)` and `Delete Key(s)`.)

I also missed this a few days ago. I guess it would help to always display all four options when right-clicking, independent of what is currently opened in the inspector.

**Comment by hsandt** [comment]
> It seems that this is only available when `AnimationTrackKeyEdit` is opened in the inspector. Otherwise, there is only `Insert Key`. (Same with `Duplicate Key(s)` and `Delete Key(s)`.)
> 
> I also missed this a few days ago. I guess it would help to always display all four options when right-clicking, independent of what is currently opened in the inspector.

Maybe we should open a bug for that. I'm right in the middle of writing another animation-related bug now, but any of you can give it a go if I haven't done it yet by then.

**Comment by KoBeWi** [comment]
> It seems that this is only available when AnimationTrackKeyEdit is opened in the inspector. Otherwise, there is only Insert Key. (Same with Duplicate Key(s) and Delete Key(s).)

No, this option only appears when you have a key selected (AnimationTrackKeyEdit doesn't need to be in the inspector). This is because the RESET value is created from existing key. If you want to add arbitrary value, open RESET animation and add a keyframe manually.

**Comment by hsandt** [comment]
> No, this option only appears when you have a key selected (AnimationTrackKeyEdit doesn't need to be in the inspector). This is because the RESET value is created from existing key

This makes sense, but the issue is that when user has *no previous key selected* and *immediately right-clicks on a key*, the "Send Key(s) to RESET" menu item will not appear, although a key is selected at that time. Looks like the logical check only checks the *previous key selection* instead of the new one we just did.

So user must first select either the same or some unrelated key to trigger the menu item, and then right-click on the actual key they want to send to RESET, which is unintuitive.

In fact since the feature I requested exists, but this odd behavior must be a bug, I opened a bug: https://github.com/godotengine/godot/issues/117100

and I'm closing this proposal now.
